crcmocks/rbac.py

```diff
@@ -1,33 +1,72 @@
+from crcmocks.config import DEFAULT_PERMISSIONS
 
 
 blueprint = Blueprint("rbac", __name__)
 
-rbac_response = {
-    "meta": {"count": 30, "limit": 1000, "offset": 0},
+BASE_RBAC_RESPONSE = {
+    "meta": {"count": None, "limit": None, "offset": 0},
     "links": {
         "first": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
         "next": None,
         "previous": None,
         "last": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
     },
-    "data": [
-        {"resourceDefinitions": [], "permission": "advisor:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-    ],
+    "data": [],
 }
 
 
-# TODO: allow customizable rbac response per user
-@blueprint.route("/v1/access/")
+# TODO: support resource definitions?
+@blueprint.route("/v1/access/", methods=["GET"])
 def rbac_access():
+    application = request.args.get("application")
+    limit = int(request.args.get("limit", 100))
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # get the user permissions
+    if user[0].get("entitlements") is not None:
+        permissions = user[0]["permissions"].split(",")
+    else:
+        permissions = DEFAULT_PERMISSIONS.copy()
+
+    rbac_response = deepcopy(BASE_RBAC_RESPONSE)
+    rbac_response["meta"]["count"] = len(permissions)
+    rbac_response["meta"]["limit"] = limit
+
+    count = 1
+    for perm in permissions:
+        if application:
+            if perm.split(":")[0] in application:
+                rbac_response["data"].append(
+                    {"resourceDefinitions": [], "permission": perm}
+                )
+                count += 1
+        else:
+            rbac_response["data"].append(
+                {"resourceDefinitions": [], "permission": perm}
+            )
+            count += 1
+        if count > limit:
+            break
+
```

Comment:
I think you'll want to edit `rbac_response["links"]["first"]` and `"last"` here:
```python
app = application or ""
rbac_response["links"]["first"] = f"/api/rbac/v1/access/?application={app}&format=json&limit={limit}&offset=0"
rbac_response["links"]["last"] = f"/api/rbac/v1/access/?application={app}&format=json&limit={limit}&offset=0"
```

Reply:
Maybe at this point it is no longer worth keeping the BASE_RBAC_RESPONSE up top since you have to edit practically all of it :laughing:

Reply:
Oh right, let me update that